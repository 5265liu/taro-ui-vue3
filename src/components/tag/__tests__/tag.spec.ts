import { mount } from '@vue/test-utils'
import AtTag from '../index'

const factory = (values = {}, slots = { default: ['标签'] }) => {
  return mount(AtTag as any, {
    slots,
    props: { ...values },
  })
}

describe('AtTag Snap', () => {
  it('should render AtTag -- props size', () => {
    const wrapper = factory({ size: 'small' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtTag -- props type', () => {
    const wrapper = factory({ type: 'primary' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtTag -- props name', () => {
    const wrapper = factory({ name: 'tag-01' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtTag -- props circle', () => {
    const wrapper = factory({ circle: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtTag -- props active', () => {
    const wrapper = factory({ active: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtTag -- props disabled', () => {
    const wrapper = factory({ disabled: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtTag Event', () => {
  it('AtTag onClick', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick })
    wrapper.find('.at-tag').trigger('tap')
    expect(onClick).toBeCalled()
  })

  it('AtTag onClick not to be called(disabled)', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick, disabled: true })
    wrapper.find('.at-tag').trigger('tap')
    expect(onClick).not.toBeCalled()
  })

  it('AtTag onClick params {name, active}', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick, name: 'tag-01' })
    wrapper.find('.at-tag').trigger('tap')
    expect(onClick.mock.calls[0][0].name).toEqual('tag-01')
    expect(onClick.mock.calls[0][0].active).toBeFalsy()
  })
})
