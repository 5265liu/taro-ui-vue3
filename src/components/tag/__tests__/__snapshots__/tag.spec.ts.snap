// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTag Snap render AtTag -- props active 1`] = `
<view
  class="at-tag at-tag--normal at-tag--active"
>
  标签
</view>
`;

exports[`AtTag Snap render AtTag -- props circle 1`] = `
<view
  class="at-tag at-tag--normal at-tag--circle"
>
  标签
</view>
`;

exports[`AtTag Snap render AtTag -- props disabled 1`] = `
<view
  class="at-tag at-tag--normal at-tag--disabled"
>
  标签
</view>
`;

exports[`AtTag Snap render AtTag -- props name 1`] = `
<view
  class="at-tag at-tag--normal"
>
  标签
</view>
`;

exports[`AtTag Snap render AtTag -- props size 1`] = `
<view
  class="at-tag at-tag--small"
>
  标签
</view>
`;

exports[`AtTag Snap render AtTag -- props type 1`] = `
<view
  class="at-tag at-tag--normal at-tag--primary"
>
  标签
</view>
`;
